### PR TITLE
RUMM-692 Fix RUM events timing

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Fix the RUM events timing.

### Motivation

With each RUM raw event (start X, stop Y, …), we keep an `eventTime` value to know when the event occurred in the client's code. At some point, those raw events could be updated to include additional attributes, and since they are all immutable objects, we used the `copy()` method.

The main issue was that the `eventTime` attribute was part of the parent `sealed` class (`RumRawEvent`), and was not copied but recreated at the time of the copy, meaning that all our events' times were skewed by a few milliseconds, making each information slightly off.

This PR fixes that by making the parent `eventTime` attribute abstract, forcing implementations to have them (meaning that they are taken into account by the kotlin compiler, and indeed kept during a `copy()` call).

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt [hunk 1]
```diff
@@ -122,13 +121,7 @@ internal class RumResourceScope(
...
 
         val finalTiming = timing
-        val duration = if (finalTiming == null) {
-            eventTime.nanoTime - startedNanos
-        } else {
-            // ensure we don't get weird graphs
-            // TODO RUMM-692 find out why we can have resource duration < timing.totalDuration()
-            max(eventTime.nanoTime - startedNanos, finalTiming.totalDuration())
-        }
+        val duration = eventTime.nanoTime - startedNanos
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:124]: No need for this check anymore, now that the timing is fixed ;)

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/resource/RumResourceInputStream.kt [hunk 2]
```diff
@@ -166,7 +164,7 @@ class RumResourceInputStream(
     // endregion
 
     companion object {
-        internal const val METHOD: String = "STREAM"
+        internal const val METHOD: String = "GET"
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/resource/RumResourceInputStream.kt:167]: We need to keep a vaild Method for now

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt (hunk 1 above)
Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:124]: You don't need the anymore the `totalDuration` function in the `RumResourceTiming` class

File: sample/README.md [hunk 3]
```diff
@@ -7,12 +7,14 @@ For each flavor you will have to provide a config file named `[flavorName].json`
...
-     "rumEndpoint": "YOUR PREFERRED ENDPOINT",
-     "token": "YOUR APP TOKEN",
-     "appId": "YOUR RUM APP ID"
-   }
+    {
+        "logsEndpoint": "YOUR PREFERRED ENDPOINT",
+        "tracesEndpoint": "YOUR PREFERRED ENDPOINT",
+        "rumEndpoint": "YOUR PREFERRED ENDPOINT",
+        "token": "YOUR APP TOKEN",
+        "rumApplicationId": "YOUR RUM APPLICATION ID",
```
Comment (mariusc83) [inline sample/README.md:15]: tks for updating this.

**Review by mariusc83 — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:124]: Indeed, I'll remove it

File: sample/README.md (hunk 3 above)
Comment (xgouchet) [reply sample/README.md:15]: ;)

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**